agreement_dynamic/models/header.py

```diff
@@ -0,0 +1,22 @@
+class Header:
```

Comment:
I've tested and the expression that I now have to use is:

```
${h1.next} New chapter
${h1.value}.${h2.next} New subchapter
${h1.value}.${h2.value}.${h3.next} New sub-subchapter
${h1.value}.${h2.value}.${h3.next} Another sub-subchapter
${h1.value}.${h2.next} New subchapter
${h1.value}.${h2.value}.${h3.reset or ''}${h3.next} New sub-subchapter
${h1.value}.${h2.value}.${h3.next} Another sub-subchapter
```

This is quite logical, but one thing that bothers me is that it requires the `h3.reset` to be mentioned explicitly.

What we could do to solve this is to define them like this:

```
h1 = Header(h2 = Header(h3 = Header()))
```

So that each will know which one is its child. If `next()` is called then it will automatically reset the children counters.

Reply:
I'll need to check out how this works